docs/source/instruction_set_extensions.rst

```diff
@@ -20,17 +20,22 @@
 CORE-V Instruction Set Extensions
 =================================
 
-CV32E40P supports the following CORE-V ISA Extensions, which are part of **Xcorev** and can be enabled by setting ``PULP_XPULP`` == 1.
+CV32E40P supports the following CORE-V ISA Extensions, which can be enabled by setting ``PULP_XPULP`` == 1.
 
- * Post-Incrementing load and stores, see :ref:`corev_load_store`.
- * Hardware Loop extension, see :ref:`corev_hardware_loop`.
- * ALU extensions, see :ref:`corev_alu`.
- * Multiply-Accumulate extensions, see :ref:`corev_multiply_accumulate`.
- * Single Instruction Multiple Data (aka SIMD) extensions, see :ref:`corev_simd`.
+ * Post-Incrementing load and stores, see :ref:`corev_load_store`, invoked in the tool chain with ``-march=xcvmem``.
+ * Hardware Loop extension, see :ref:`corev_hardware_loop`, invoked in the tool chain with ``-march=xcvhwlp``.
+ * ALU extensions, see :ref:`corev_alu`, which are divided into three sub-extensions:
 
-Additionally the event load instruction (**cv.elw**) is supported by setting ``PULP_CLUSTER`` == 1.
+   * bit manipulation instructions, invoked in the tool chain with ``-march=xcvbitmanip``;
+   * miscellaneous ALU instructions, invoked in the tool chain with ``-march=xcvalu``; and
+   * immediate branch instructions, invoked in the tool chain with ``-march=xcvbi``.
 
-To use such instructions, you need to compile your SW with the CORE-V GCC compiler.
+ * Multiply-Accumulate extensions, see :ref:`corev_multiply_accumulate`, invoked in the tool chain with ``-march=xcvmac``.
+ * Single Instruction Multiple Data (aka SIMD) extensions, see :ref:`corev_simd`, invoked in the tool chain with ``-march=xcvsimd``.
+
+Additionally the event load instruction (**cv.elw**) is supported by setting ``PULP_CLUSTER`` == 1, see :ref:`corev_event_load`.  This is a separate ISA extension, invoked in the tool chain with ``-march=xcvelw``.
+
+To use such instructions, you need to compile your SW with the CORE-V GCC or Clang/LLVM compiler.
```

Comment:
Maybe we should specify starting from which version those instructions are supported by the toolchain, GCC and LLVM.

Reply:
A good suggestion. I'll make this change.